documentation/adrs/001-documentation-tool.md

```diff
@@ -0,0 +1,69 @@
+# Choosing a Documentation Tool
+
+We want to offer exemplary DevTK documentation for current and future users to learn and discover the functionalities contained.
+
+## Goals
+
+  1. Docblocks inside the codebase
+      - These are inline comments documenting the structure, usage, and behavior of specific functions, classes, and modules.
+      - Docblocks are helpful for users and maintainers as they provide information in close to where the code is being used.
+  2. Markdown outside of the codebase
+      - Markdown files outside the codebase can contain additional documentation such as high-level architectural decisions, design intentions, and contextual explanations.
+  3. Easy maintenance
+      - There are at least two aspects to maintenance: maintaining the documentation to keep it accurate, and generating the tool used to browse the documentation.
+      - The Docblocks are helpful but including exhaustive documentation in the codebase will become burdensome and avoided.
+      - Maintaining two (or more) sources of truth for any content will mean that it will become out of date very easily and quickly; the only thing worse than no documentation is bad/incorrect documentation.
+      - We will have plenty of work building the modules we don't need the additional responsibility of keeping a whole other project - the documentation site - up to date and accurate.
+  4. "No-setup" playgrounds
+      - Where possible, for each module, there could be an interactive component that would allow users to experiment with the codebase from within the documentation site itself.
+  5. Limits
+      - dont want to replace training suite
+      - dont want to replace code comments
```

Comment:
I made a comment in the chat about this but I think it is worth addressing audience as a part of the requirements.  It is a nice sanity check on decisions that get made to ensure that multiple perspectives and roles are getting addressed and can access/use/contribute to the docs. 

- Designers 

Making the systems more transparent and consolidated will allow designers to improve and integrate new designs and make efforts more efficient and targeted 

- Application Developers (Jeric2o, Hawkeye, etc) 
As the consumers of the platform resources, enabling easy access and clear documentation on how to leverage what is available will help developer velocity and adoption. 

- Platform Developers 
As builders of the platform, devs will have a clear channel for explaining how and why the resources should be used, reducing mental load on communication strategies 

- Onboarding Developers and Designers 
Onboarding people will be simpler because there will be a consolidated way to communicate a lot of context and the ability to reference important materials 

- Product Owners and Managers 
Access to understanding the up to date capabilities makes it easier to plan and integrate the tools that are available into the existing product systems 

- Business Stakeholders 
By having a clear overview of the capabilities and systems that are in usage, it is more possible to see the 30,000 foot view and conceptualize places to scale and grow

Reply:
What are/were your thoughts around "worth making sure whatever is chosen works across the board even for non-coder types"? Do you mean the documentation content (end result) or more the editing and maintaining of the content?

Reply:
I was referring to contributions specifically. The technical bits of the docs are important, but ultimately the docs are only as good as the content...and a lot of the content will be "business" content that is owned by designers or potentially technical writers. But it looks like most of these use MDX or markdown text files so that seems doable.

Reply:
Markdown (and similar i.e. MDX) was an unofficial requirement for me I think.